draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright [2022] [Application library engineering group]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+import Curves from '@ohos.curves'
+import { Cell } from './DraggableTypes'
+import { DraggableModel } from './DraggableModel'
+
+const springSetting1: string = Curves.spring(0, 1, 180, 10)
+const springSetting2: string = Curves.spring(0, 1, 120, 15)
+
+@Component
+export struct Draggable {
+  model: DraggableModel.Model = undefined
+  private mouseXY: number[] = [0, 0]
+  private positionOffsetCorrection: number[] = [0, 0]
+  @State private lastPress: number = undefined
+  @State private isPressed: boolean = false
+  @State private scale: number = 1
+  @State private order: number[] = undefined
+  private innerFlexSize: SizeOptions = undefined
+  private layout: Cell[] = undefined
+  private ballCount: number
+  private cellWidth: number
+  private cellHeight: number
+
+  aboutToAppear() {
+    if(this.model == undefined || this.model == null) {
+      console.error('Please pass correct model to the Draggable component');
+      return;
+    }
+    this.ballCount = this.model.getColorsList().length
```

Comment:
getColorsList() values can be null as user is passing it... please recheck for all such validations